packages/core/src/utilities/RLEVoxelMap.ts

```diff
@@ -0,0 +1,156 @@
+import isEqual from './isEqual';
+
+export type RLERun<T> = {
+  value: T;
+  i: number;
+  iEnd: number;
+  run: RLERun<T>;
+};
+
+/**
+ * RLE based implementation of a voxel map.
+ * This can be used as single or multi-plane, as the underlying indexes are
+ * mapped to rows and hte rows are indexed started at 0 and continuing
+ * incrementing for all rows in the multi-plane voxel.
+ */
+export default class RLEVoxelMap<T> {
+  protected rows = new Map<number, RLERun<T>>();
+  protected height = 1;
+  protected width = 1;
+  protected depth = 1;
+  protected jMultiple = 1;
+  protected kMultiple = 1;
+
+  constructor(width: number, height: number, depth = 1) {
+    this.width = width;
+    this.height = height;
+    this.depth = depth;
+    this.jMultiple = width;
+    this.kMultiple = this.jMultiple * height;
+  }
+
+  public get = (index): T => {
+    const i = index % this.jMultiple;
+    const j = (index - i) / this.jMultiple;
+    const row = this.rows.get(j);
+    if (!row) {
+      return;
+    }
+    for (let run = row; run; run = run.run) {
```

Comment:
what is run and then we have run.run

Reply:
Removed that - now have row and rle, with a type for RLE.